utils: fix build and add a method to target specific props and extend (#329)
BuilderIO/builder: packages/utils/README.md

  return { props: { content } }
}
```

## extendAsyncProps

> `extendAsyncProps` is similar to `getAsyncProps` except it look up the inputs by name across components and map it's props to the result of the asynchronus mapper function passed.

Reviewer: ```suggestion
`extendAsyncProps` is similar to `getAsyncProps` except it looks up the inputs by name across components and map its props to the result of the asynchronus mapper function passed.
```
Also, maybe reword this in simpler terms?

> map its props to the result of the asynchronus mapper function passed.

It might takes a few re-reads to understand properly...

### Example

> Say you have multiple components that takes a query input and you want to resolve this query asynchronusly before rendering on the server

Reviewer 2: ```suggestion
Say you have multiple components that take a query input and you want to resolve this query asynchronusly before rendering on the server
```

    query: 'shirt',
  },
  subFields: [
    {
      type: 'string',
      name: 'query',
      helperText: 'for syntax check https://shopify.dev/concepts/about-apis/search-syntax',
    },
    {
      type: 'enum',
      name: 'sortBy',
      enum: [
        {
          label: 'product ID',
          value: 'ID',
        },
        {
          label: 'Date of creation',
          value: 'CREATED_AT',
        },
      ],
    },
    {
      type: 'number',
      name: 'first',
    },
    {
      type: 'number',
      name: 'last',
    },
    {
      type: 'boolean',
      name: 'reverse',
    },
  ],
}
Builder.registerComponent(Products, { 
  name: 'ProductGrid', inputs: [customQueryInput}]
})
// another component 
Builder.registerComponent(ProductsSlider, { 
  name: 'ProductsSlider', inputs: [customQueryInput}]
})

```
This component will update to fetch product data browser side by query chosen in the Builder editor. But if you want this to render server side you need `props.products` to be passed with the needed data on render.

To accomplish this, you can use `extendAsyncProps`. 

Here is a usage example with Next.js:

```tsx
import { Builder, builder } from '@builder.io/react';

> import { extendAsyncProps } from '@builder.io/utils';

Reviewer: In this example, this is imported, but not used?

export default function MyPage(props) {
  return <BuilderComponent model="page" content={props.content} />
}

export async function getStaticProps(context) {
  const content = await builder.get('page', { url: context.resolvedUrl }).promise();
  

>   await getAsyncProps(content, {

Reviewer: ```suggestion
  await extendAsyncProps(content, {
```
Is this what you mean?
